### DISCUSSION TITLE
Get all Instances

### DISCUSSION THREAD

**Discussion by mrousavy** [Tencent/MMKV#992] [General]
Hey!

I couldn't find any information in the docs or in the header file on how one could get a list of all MMKV instances (or a list of all IDs) - is that possible?

Thanks!

**Comment by lingol** [comment]
There's no such interface. Why'd you like to get all instances?

**Comment by mrousavy** [comment]
I want to delete/nuke the entire MMKV storage - basically delete all instances.

**Comment by lingol** [comment]
Do you want to delete live instances or the underlying file storage?

**Comment by mrousavy** [comment]
I want to delete both the file storage and all instances that are in memory. Basically reset everything to 0.

**Comment by mrousavy** [comment]
Hey @lingol, sorry for bumping this but does MMKV have such a feature?

**Comment by lingol** [comment]
No.

**Reply by mrousavy** [reply]
Okay, do you have plans to support this or are you open to accept PRs that add support for this?

**Reply by lingol** [reply]
We don't have any plans for it. PRs are always welcome. 

However I should mind you it's tricky to provide such a functionality. Maybe the user has not created all instances before calling the method. Or worse, maybe some of these instances are encrypted in file.
